webextension/textarea.js

```diff
@@ -0,0 +1,432 @@
+  const popupHeight = Math.min(window.innerHeight * 80 / 100, 600);
+  $.featherlight({
+    iframe: `${chrome.runtime.getURL("popup.html")}?pageUrl=${currentUrl}`,
+    iframeWidth: popupWidth,
+    iframeHeight: popupHeight,
+    namespace: "ltaddon-popup",
+    beforeOpen: () => {
+      const popupContainers = document.getElementsByClassName(
+        POPUP_CONTENT_CLASS
+      );
+      for (let counter = 0; counter < popupContainers.length; counter += 1) {
+        const popupContainer = popupContainers[counter];
+        popupContainer.style.minWidth = `${popupWidth}px`;
+        popupContainer.style.minHeight = `${popupHeight}px`;
+      }
+    },
+    afterOpen: () => {
+      const currentPopup = $.featherlight.current();
+      currentPopup.$content.focus();
+    }
+  });
+}
+
+function removeAllButtons() {
+  const btns = document.getElementsByClassName(REMIND_WRAPPER_CLASS);
+  for (let counter = 0; counter < btns.length; counter += 1) {
+    const btn = btns[counter];
+    btn.parentNode.removeChild(btn);
+  }
+}
+
+function disableMenu(evt) {
+  evt.preventDefault();
+  disableOnDomain = true;
+  removeAllButtons();
+  Tools.getStorage().get(
+    {
+      disabledDomains: []
+    },
+    items => {
+      const currentUrl = window.location.href;
+      const { hostname } = new URL(currentUrl);
+      items.disabledDomains.push(hostname);
+      Tools.getStorage().set({
+        disabledDomains: [...new Set(items.disabledDomains)]
+      });
+    }
+  );
+}
+
+/** DOM manupulate */
+function remindLanguageToolButton(clickHandler, position) {
+  const { top, left, offsetHeight, offsetWidth } = position;
+  const btn = document.createElement("A");
+  btn.onclick = clickHandler;
+  btn.className = `${BTN_CLASS} ${REMIND_BTN_CLASS}`;
+  btn.setAttribute("tooltip", chrome.i18n.getMessage("reminderIconTitle"));
+
+  // style
+  btn.style.position = "absolute";
+  if (isGmail()) {
+    const tables = document.querySelectorAll("table#undefined");
+    const activeTable = Array.prototype.find.call(tables, table =>
+      isDescendant(table, document.activeElement)
+    );
+    // find parent of active table
+    const allTables = document.getElementsByTagName("table");
+    const gmaiComposeToolbarHeight = 155;
+    for (let counter = allTables.length - 1; counter > 0; counter -= 1) {
+      const parentTable = allTables[counter];
+      if (isDescendant(parentTable, activeTable)) {
+        let topPostion = offset(parentTable).top;
+        if (topPostion < gmaiComposeToolbarHeight) {
+          topPostion = gmaiComposeToolbarHeight;
+        }
+        btn.style.top = `${topPostion}px`;
+        break;
+      }
+    }
+  } else {
+    btn.style.top = `${top +
+      offsetHeight -
+      REMIND_BTN_SIZE -
+      MARGIN_TO_CORNER}px`;
+  }
+  btn.style.left = `${left +
+    offsetWidth -
+    REMIND_BTN_SIZE -
+    MARGIN_TO_CORNER}px`;
+
+  return btn;
+}
+
+function disableLanguageToolButton(clickHandler, position) {
```

Comment:
There's some code duplication here with `remindLanguageToolButton`, could this be avoided?

Reply:
ok, let me improve this